### ISSUE TITLE
Child Pages / Project Repo Deployment

### ISSUE THREAD

**Issue by cmharlow** [DLFMetadataAssessment/Sandbox#20]
Figure out how to deploy:

- [ ]  for "development" (to Heroku)
- [ ]  then for "production" (so to dlfmetadataassessment.github.io/environmental-scan)

See: https://help.github.com/enterprise/2.10/user/articles/user-organization-and-project-pages/

**Comment by kateefly** [comment]
Question for the group at large - would it be ok to work with the EnvScan repository as is or should I update it to include what's currently in Sandbox? I think it's a couple of commits off, some of which was structural rather than content. This is something to likely come up again so I figured it's good to have the discussion/documentation happen now rather than later.

**Comment by cmh2166** [comment]
- Absolute back up plan: publish / deploy as GitHub project pages to production URL
- Heroku configurations for their application domains / urls?
- Leave navigation to end?
- Document placement of links.

**Comment by kateefly** [comment]
Crazy, untested idea that I think may be our best bet - should we pull in child repos as submodules of the Sandbox directory? Heroku will support submodules so it just might work. I also took a look around at blogs and Stack Overflow but wasn't able to find someone who had done exactly this. So we can't be sure if it'll work/what will break until we try it. Another downside is that we'll also have to undo the submodule set-up before we transfer Sandbox changes to the production site (since we would otherwise be duplicating child repositories). 

Here's the documentation on submodules from Heroku: https://devcenter.heroku.com/articles/git-submodules
And from Git: https://git-scm.com/book/en/v2/Git-Tools-Submodules

What does everyone think?

**Comment by billmcmillin** [comment]
I agree that submodules seem like our best bet. 

I created a Pipeline in Heroku that takes PRs, launches them as Preview Apps, then destroys the Preview App after 5 days. So for this one, I added EnvironmentalScan as a submodule, then deployed to https://dlfaigmwgdev-pr-37.herokuapp.com/

Then saw: "Using submodules for builds on Heroku is only supported for builds triggered with git pushes. Builds created with the API do not resolve submodules. This is also the case for GitHub sync."  So the app launched, but without the submodule.

So it looks like the submodule will not be built in because we're using GitHub sync. I'm looking for a workaround.

**Comment by billmcmillin** [comment]
Preview apps won't work, but we can deploy from the master branch when using Heroku git. Once branch 'issue#20' is merged, we'll know how Heroku handles submodule deploys from the master branch on GH. If this works as expected, we should be ready to test production deploy.

**Comment by kateefly** [comment]
Development deployment is on hold. Focusing on production deployment in prep for DLF Forum

**Comment by kateefly** [comment]
@kateefly will deploy EnvScan & Tools. Other child repos have been deployed to production.
